packages/web-components/src/theme/set-theme.ts

```diff
@@ -0,0 +1,12 @@
+import type { Theme } from '@fluentui/react-theme';
+import * as tokens from './design-tokens';
+
+/**
+ * Sets the theme tokens on defaultNode.
+ * @param theme Flat object of theme token values.
+ */
+export const setTheme = (theme: Theme) => {
+  for (const t of Object.keys(tokens)) {
```

Comment:
can we extract `Object.keys(tokens)` outside this function ? this might add some unwanted processing time (laggy UI)

Reply:
It would be good to bench that to be sure. I think `Object.keys` should just return an iterator which for/of should hold on to. I don't think there should be an issue.

Reply:
https://jsbench.me/w8lb5gricd/1

Reply:
Yep. Difference looks negligible.

Reply:
Oops actually I think I misunderstood the case. https://jsbench.me/w8lb5gricd/2 seems in-line with the suggested refactor, which appears to be faster in Edge.

Reply:
Ah! Yes. That makes sense.